src/components/studies/participants/BatchEditForm.tsx

```diff
@@ -0,0 +1,95 @@
+import React from 'react'
+import { Dialog } from '@material-ui/core'
+import { EditParticipantForm } from './ParticipantForms'
+import { EditDialogTitle } from './ParticipantTableGrid'
+import ParticipantService from '../../../services/participants.service'
+
+type BatchEditFormProps = {
+  isEnrolledById: boolean
+  isBatchEditOpen: boolean
+  setIsBatchEditOpen: Function
+  selectedParticipants: string[]
+  token: string
+  studyId: string
+  toggleParticipantRefresh: Function
+  isAllSelected: boolean
+}
+
+const BatchEditForm: React.FunctionComponent<BatchEditFormProps> = ({
+  isEnrolledById,
+  isBatchEditOpen,
+  setIsBatchEditOpen,
+  selectedParticipants,
+  token,
+  studyId,
+  toggleParticipantRefresh,
+  isAllSelected,
+}) => {
+  const [isLoading, setIsLoading] = React.useState(false)
+
+  return (
+    <Dialog
+      open={isBatchEditOpen}
+      maxWidth="sm"
+      fullWidth
+      aria-labelledby="edit participant"
+    >
+      <EditDialogTitle
+        onCancel={() => {
+          setIsBatchEditOpen(false)
+        }}
+        shouldWithdraw={false}
+        batchEdit
+      />
+      <EditParticipantForm
+        isEnrolledById={isEnrolledById}
+        onCancel={() => {
+          setIsBatchEditOpen(false)
+        }}
+        onOK={async (note: string, cvd?: Date) => {
+          if (!note && !cvd) return
+          setIsLoading(true)
+          let participants = selectedParticipants
+          if (isAllSelected) {
+            const resultEnrolled = await ParticipantService.getAllParticipantsInEnrollmentType(
+              studyId,
+              token,
+              'enrolled',
+              false,
+            )
+            const enrolledNonTestParticipants = resultEnrolled.items
```

Comment:
The variable naming on ln54 and 60 are a little misleading. Ln60 should be same as 54 but just Id's right?

Reply:
Line 60 is the entire participant object, not just ids. Line 54 is an object that has the participant data but also the total number of participants (which is not needed in our case).